astropy/units/quantity.py

```diff
@@ -1451,25 +1462,39 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
             },
         }
         formats["latex_inline"] = formats["latex"]
+        default_formatter = None
```

Comment:
I think this can still be moved inside the `if format_spec is None` branch. I.e., something like
```
if format is None:
    if format_spec is None:
        if precision is None:
            return ...
        default_formatter = None
    else:
        default_formatter = {"all": lambda x: format(x, format_spec)}
```
Note the change to use `format` - it should be faster and is more explicit.